### DISCUSSION TITLE
How to hide the tooltip when the mouse moves and display it again.

### DISCUSSION THREAD

**Discussion by CauePizzol** [AvaloniaUI/Avalonia#8861] [Q&A] [answered]
Hello,
I have a textbox with a tooltip. The current behavior makes the tooltip visible when the mouse is hovered over the text box, which is expected. However, if I move my mouse, within that same element, the tooltip keeps it's position. Ideally, I would like it to disappear if the mouse started moving and reappear after the mouse stops (if in an element that contains a tooltip).
Gif of current behavior:
![Tooltip problem](https://user-images.githubusercontent.com/112198065/187469297-a86131b3-ab3f-408b-946a-10c138605d66.gif)

Thanks!

**Comment by timunie** [comment, marked as answer]
ToolTip has attached property IsOpen which you can set for any control. See #8833. You can turn it into a behavior which can be reused on any control. 

Happy coding
Tim

**Reply by CauePizzol** [reply]
Hello, 
thanks for your reply.
My issue here isn't exactly with "isOpen" property, but rather how to update the tooltip.
I tried using "OnPointerMoved" and setting the property IsOpen to false and then to true, but that didn't work.
I also tried setting the offset by +1 pixel and -1 pixel every time "OnPointerMoved" but that give it a weird blurry sensation, although it did work to update the tooltip position. 
Is there an update position property that I can use, that I am missing?
Thanks again :)

**Reply by timunie** [reply]
Well, that's a different Story. The original question was to close and open the ToolTip. As I was curious as well I played around a bit with it and came up with this idea: 

in XAML: 
```xml
<TextBlock Text="{Binding Greeting}" Background="Transparent"
           ToolTip.Tip="Hello World"
           ToolTip.Placement="Top"
           PointerEntered="InputElement_OnPointerEnter"
           PointerMoved="InputElement_OnPointerMoved"
           HorizontalAlignment="Stretch"
           VerticalAlignment="Stretch"/>
```

And in code behind I calculate the position on my own: 

```cs
private void InputElement_OnPointerMoved(object? sender, PointerEventArgs e)
{
    if (sender is Control ctrl)
    {
        ToolTip.SetHorizontalOffset(ctrl, e.GetPosition(ctrl).X + 10);
        ToolTip.SetVerticalOffset(ctrl, e.GetPosition(ctrl).Y + 10);
    }
}

private void InputElement_OnPointerEnter(object? sender, PointerEventArgs e)
{
    if (sender is Control ctrl)
    {
        ToolTip.SetHorizontalOffset(ctrl, e.GetPosition(ctrl).X + 10);
        ToolTip.SetVerticalOffset(ctrl, e.GetPosition(ctrl).Y + 10);
    }
}
```

-----

On the other hand it may be great to have a ToolTip which is following the pointer automatically. I don't know if that is something which can (or should) be implemented in Avalonia directly.

**Reply by CauePizzol** [reply]
Thanks for your answer. 
I had tried something similar and it doesn't work for me. The tooltip position never gets updated (same behavior as the first gif).
What somewhat worked for me was this inside the "OnPointerMoved":
```
if(ToolTip.GetHorizontalOffset(ctrl) < 15)
{
	ToolTip.SetHorizontalOffset(_ctrl, 15);
}
else
{
	ToolTip.SetHorizontalOffset(ctrl, 14);
}
```
But for me, this is not the ideal solution as "OnPointerMoved" keeps getting called even when the mouse is standing still, which gives the tooltip a blurry sensation.

![Tooltip](https://user-images.githubusercontent.com/112198065/188131039-bfe57dc5-366a-4fad-b68e-c8b8a4f0199a.gif)

I don't know if Avalonia should implement something like this directly, but if there was a way to update the tooltip position, instead of just the offset, it would be great.

**Reply by timunie** [reply]
Have you double checked that position is not Pointer in my sample but Top?

**Reply by CauePizzol** [reply]
I'm not sure. I tried your suggestion and it didn't update the position. How can I check that?
Thanks

**Reply by timunie** [reply]
Please provide a small sample App of your impl. I'll have a look then.

**Reply by timunie** [reply]
Can you please double-check if you have this line set? `ToolTip.Placement="Top"` (In XAML)

**Reply by CauePizzol** [reply]
Hey, I don't have it. I'll give it a try and comeback to you.
Thanks :)

**Reply by CauePizzol** [reply]
You are right! My tooltip placement was on Pointer. This solved my issue. 
Thanks 👍
